### DISCUSSION TITLE
How can I use the same model's field for more than 1 column

### DISCUSSION THREAD

**Discussion by cleargoal** [filamentphp/filament#12550] [Help] [answered]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

I try code like this:
```
    IconColumn::make('confirmed')->label('Зарахування')->width('5rem')->alignment(Alignment::Center)
        ->sortable()
        ->icon(fn(bool $state): string => match ($state) {
            true => 'heroicon-o-check-circle',
            false => 'heroicon-o-clock',
        })
        ->color(fn(bool $state): string => match ($state) {
            true => 'success',
            false => 'warning',
        }),
    ToggleColumn::make('confirmed')->label('Підтвердження')->alignment(Alignment::Center)
        ->afterStateUpdated(function ($record, $state) {
            (new TotalCalculator())->processing($record);
        }),
```
But only ToggleColumn there is in the table. Otherwise, before I added ToggleColumn, the IconColumn was shown well.

IconColumn just needs to be shown and has no functionality, so it can't break anything.

The matter is that **these 2 columns need to be shown to users with different roles** and my next challenge is to add role control to this resource.

Any help and suggestions are welcome!

**Comment by leandrocfe** [comment, marked as answer]
try this..

```php
IconColumn::make('confirmed_icon')
->state(fn(?YourModel $record) => $record->confirmed)
->icon(...)
->color(...)
...
```

**Reply by cleargoal** [reply]
Thank you!

Nothing changed

**Reply by leandrocfe** [reply]
Did you change the column name to confirmed_icon?
Could you share a screenshot showing the problem?

**Reply by cleargoal** [reply]
Oh, sorry!
I didn't notice that it needed to be changed.
I changed it and now both columns are shown.
**Thank you very much!**
